include/utility/json_utility.hpp

```diff
@@ -650,5 +650,132 @@ inline bool isBackupAndRestoreRequired(const nlohmann::json& i_sysCfgJsonObj)
+    }
+
+    return false;
+}
+
+/**
+ * @brief API to return parameters required for polling.
+ *
+ * This API checks for "gpioPresence" tag under "pollingRequired" in system
+ * config JSON. If present returns the parameters for GPIO polling.
+ *
+ * @throw JSON Exception, if JSON does not contain "pollingRequired" tag under
+ * the given VPD file path.
+ *
+ * @param[in] i_sysCfgJsonObj -  System config JSON object.
+ * @param[in] i_vpdFilePath - Path to where VPD is present.
+ *
+ * @return GPIO parameters if present, otherwise empty tuple.
+ */
+inline types::GpioPollingParameters
+    getGpioPollingParameters(const nlohmann::json& i_sysCfgJsonObj,
+                             const std::string i_vpdFilePath)
+{
+    types::GpioPollingParameters l_pollingParameters;
+
+    if (i_sysCfgJsonObj.empty() || i_vpdFilePath.empty())
+    {
+        logging::logMessage("Invalid parameter");
+        return l_pollingParameters;
+    }
+
+    if (!i_sysCfgJsonObj.contains("frus"))
+    {
+        logging::logMessage("Missing frus section in config JSON");
+        return l_pollingParameters;
+    }
+
+    if (i_sysCfgJsonObj["frus"][i_vpdFilePath]
+            .at(0)["pollingRequired"]
+            .contains("gpioPresence"))
+    {
+        std::get<0>(l_pollingParameters) =
+            i_sysCfgJsonObj["frus"][i_vpdFilePath].at(
+                0)["pollingRequired"]["gpioPresence"]["pin"];
+        std::get<1>(l_pollingParameters) =
+            i_sysCfgJsonObj["frus"][i_vpdFilePath].at(
+                0)["pollingRequired"]["gpioPresence"]["value"];
+        std::get<2>(l_pollingParameters) =
+            i_sysCfgJsonObj["frus"][i_vpdFilePath].at(0)["inventoryPath"];
+    }
+
+    return l_pollingParameters;
+}
+
+/**
+ * @brief An API returns list of FRUs parameters that needs Polling.
+ *
+ * An API that checks for the FRUs that requires GPIO polling and returns
+ * a list of FRUs with polling parameters. Returns an empty list if there are
+ * no FRUs that requires polling.
+ *
+ * @param[in] i_sysCfgJsonObj - System config JSON object.
+ *
+ * @return list of FRUs parameters that needs polling.
+ */
+inline std::vector<types::GpioPollingParameters>
+    getFrusParamThatNeedsPolling(const nlohmann::json& i_sysCfgJsonObj)
+{
+    std::vector<types::GpioPollingParameters> l_frusPollingParamsList;
+
+    if (i_sysCfgJsonObj.empty())
+    {
+        logging::logMessage("Invalid Parameter");
+        return l_frusPollingParamsList;
+    }
+
+    if (!i_sysCfgJsonObj.contains("frus"))
+    {
+        logging::logMessage("Missing frus section in config JSON");
+        return l_frusPollingParamsList;
+    }
+
+    for (const auto& l_fru : i_sysCfgJsonObj["frus"].items())
+    {
+        const auto l_fruPath = l_fru.key();
+
+        if (jsonUtility::isPollingRequired(i_sysCfgJsonObj, l_fruPath))
+        {
+            types::GpioPollingParameters l_pollingParameters =
+                jsonUtility::getGpioPollingParameters(i_sysCfgJsonObj,
```

Comment:
What happens when empty tuple is returned from API "getGpioPollingParameters"?
Should we push that in the list?
Handling empty tuple is a bit tricky. Best throw in case of any error in the API "getGpioPollingParameters" and based on the throw either push that into list or skip.